core-web/libs/sdk/experiments/src/lib/hooks/DotExperimentsProvider.tsx

```diff
@@ -0,0 +1,78 @@
+import { ReactElement, ReactNode, useEffect, useState } from 'react';
+
+import { isInsideEditor } from '@dotcms/client';
+
+import DotExperimentsContext from './DotExperimentsContext';
+
+import { DotExperiments } from '../dot-experiments';
+import { DotExperimentConfig } from '../shared/models';
+
+interface DotExperimentsProviderProps {
+    children?: ReactNode;
+    config: DotExperimentConfig;
+}
+
+/**
+ * `DotExperimentsProvider` is a component that uses React's Context API to provide
+ * an instance of `DotExperiments` to all of its descendants.
+ *
+ *
+ * @component
+ * @example
+ * ```jsx
+ *
+ * // Your application component
+ * function App() {
+ *
+ * // Other configuration options will be taken from environment variables.
+ *  const config = {
+ *     'api-key': process.env.NEXT_PUBLIC_EXPERIMENTS_API_KEY ,
+ *     server: process.env.NEXT_PUBLIC_DOTCMS_HOST ,
+ *     debug: process.env.NEXT_PUBLIC_EXPERIMENTS_DEBUG,
+ *     redirectFn: YourRedirectFunction
+ *   };
+
+ *
+ *   return (
+ *     <DotExperimentsProvider config={config}>
+ *       <YourDescendantComponent />
+ *     </DotExperimentsProvider>
+ *   );
+ * }
+ * ```
+ *
+ * @param {object} props - The properties that define the `DotExperimentsProvider`.
+ * @param {ReactNode} props.children - The descendants of this provider, which will
+ *   have access to the provided `DotExperiments` instance.
+ * @param {DotExperimentConfig} props.config - The configuration object for `DotExperiments`.
+ *
+ * @returns {ReactElement} The provider component, which should wrap the components
+ * that need access to the `DotExperiments` instance.
+ */
+export const DotExperimentsProvider = ({
+    children,
+    config
+}: DotExperimentsProviderProps): ReactElement => {
+    const [instance, setInstance] = useState<DotExperiments | null>(null);
+
+    // Initialize the DotExperiments instance
+    useEffect(() => {
+        const insideEditor = isInsideEditor();
+        if (!insideEditor) {
+            const dotExperimentsInstance = DotExperiments.getInstance(config);
+            dotExperimentsInstance.ready().then(() => {
+                setInstance(dotExperimentsInstance);
+            });
+        } else {
+            if (config.debug) {
+                console.warn(
+                    'DotExperimentsProvider: DotExperiments instance not initialized because it is inside the editor.'
+                );
+            }
+        }
+    }, [config]);
+
+    return (
+        <DotExperimentsContext.Provider value={instance}>{children}</DotExperimentsContext.Provider>
+    );
```

Comment:
```suggestion
    return (
        <DotExperimentsContext.Provider value={instance}>
            {children}
        </DotExperimentsContext.Provider>
    );
```

Reply:
Let me try, or verify why the prettier force me to the one line.